lib/new_relic/agent/instrumentation/logger/instrumentation.rb

```diff
@@ -35,6 +35,7 @@ def self.enabled?
 
         def format_message_with_tracing(severity, datetime, progname, msg)
           formatted_message = yield
+          ::NewRelic::Agent.agent.log_event_aggregator.register_logger(self) unless skip_instrumenting?
```

Comment:
If this was moved below `return formatted_message if skip_instrumenting?` then it wouldn't need the conditional check at all right?

Reply:
Thanks, @jasonrclark!

I was misusing the existing skipped/cleared instance variables to see if a logger was registered yet or not. This caused the confusing duplicated conditional check here and the problematic interference with existing behavior that you noted in your other comment.

With [f32f947](https://github.com/newrelic/newrelic-ruby-agent/pull/1041/commits/f32f947a34b1c04d80c56bf029c494e21a21a632), I have reworked things so that:

a) We have a new @newrelic_registered instance variable to track if a logger has been registered
b) Non stdout/stderr loggers are no longer cleared for instrumentation, they're simply not marked for skipping.

Reply:
Totally makes sense and changes look good to me 👍